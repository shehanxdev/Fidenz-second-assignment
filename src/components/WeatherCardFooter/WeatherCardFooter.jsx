import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faLocationArrow } from "@fortawesome/free-solid-svg-icons";
import React from "react";

export function WeatherCardFooter(weatherData) {
  const customBorderStyles = {
    border: "1px solid #383B47",
  };
  return (
    <div>
      <div className="d-flex justify-content-center gap-3 p-4 bg-dark">
        <div className="d-flex flex-column">
          <span>Pressure: 1018hPa</span>
          <span>Humidity: 78%</span>
          <span>Visibility: 8.0km</span>
        </div>
        <div style={customBorderStyles}></div>
        <div className="d-flex flex-column justify-content-center">
          <FontAwesomeIcon icon={faLocationArrow} style={{ color: "white" }} />
          <span>4.0m/s 120 Degree</span>
        </div>
        <div style={customBorderStyles}></div>
        <div className="d-flex flex-column justify-content-center">
          <span>Sunrise: 6:05am</span>
          <span>Sunset: 6:05am</span>
        </div>
      </div>
    </div>
  );
}
