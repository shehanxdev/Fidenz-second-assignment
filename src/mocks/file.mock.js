export const mockWeatherData = {
  coord: { lon: -74.006, lat: 40.7128 },
  sunrise: 1642396800,
  sunset: 1642436799,
  country: "US",
  description: "Clear sky",
  icon: "01d",
  temp: 25.5,
  pressure: 1015,
  humidity: 50,
  temp_min: 22.0,
  temp_max: 28.0,
  visibility: 10000,
  speed: 5.0,
  deg: 180,
  all: 1,
  dt: 1642422236,
  id: 5128581,
  name: "New York",
};
