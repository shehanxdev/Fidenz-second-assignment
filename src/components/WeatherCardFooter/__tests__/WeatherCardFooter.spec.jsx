import React from "react";
import renderer from "react-test-renderer";
import { WeatherCardFooter } from "../WeatherCardFooter";
import { mockWeatherData } from "../../../mocks";
describe("WeatherCardFooter component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<WeatherCardFooter weatherData={mockWeatherData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
