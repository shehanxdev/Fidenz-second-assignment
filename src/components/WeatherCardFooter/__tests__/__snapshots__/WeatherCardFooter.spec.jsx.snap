// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardFooter component renders correctly 1`] = `
<div>
  <div
    className="d-flex justify-content-center gap-3 p-4 "
    style={
      Object {
        "backgroundColor": "#383B47",
      }
    }
  >
    <div
      className="d-flex flex-column"
    >
      <span>
        Pressure: 1015hPa
      </span>
      <span>
        Humidity: 50%
      </span>
      <span>
        Visibility: 10km
      </span>
    </div>
    <div
      style={
        Object {
          "border": "1px solid #585a61",
        }
      }
    />
    <div
      className="d-flex flex-column justify-content-center gap-2"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-location-arrow fa-2x "
        data-icon="location-arrow"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M429.6 92.1c4.9-11.9 2.1-25.6-7-34.7s-22.8-11.9-34.7-7l-352 144c-14.2 5.8-22.2 20.8-19.3 35.8s16.1 25.8 31.4 25.8H224V432c0 15.3 10.8 28.4 25.8 31.4s30-5.1 35.8-19.3l144-352z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        5.0m/s 180 Degree
      </span>
    </div>
    <div
      style={
        Object {
          "border": "1px solid #585a61",
        }
      }
    />
    <div
      className="d-flex flex-column justify-content-center"
    >
      <span>
        Sunrise: 6:05am
      </span>
      <span>
        Sunset: 6:05am
      </span>
    </div>
  </div>
</div>
`;
